core/src/main/java/org/mule/DefaultMuleEvent.java

```diff
@@ -272,6 +299,16 @@ public DefaultMuleEvent(MuleEvent rewriteEvent, FlowConstruct flowConstruct)
             rewriteEvent.isSynchronous());
     }
 
+    /**
+     * Copy constructor used when ReplyToHandler instance needs switching out
+     * @param rewriteEvent
+     * @param replyToHandler
+     */
+    public DefaultMuleEvent(MuleEvent rewriteEvent, ReplyToHandler replyToHandler)
```

Comment:
The variation of constructors in this class is a mess... would be good to introduce a builder in 3.7